### PR TITLE
add PyClibrary

### LINKED ISSUE #15962: [Package request] pyclibrary
# Some links to get started
- [Currently available conda-forge packages](https://conda-forge.org/feedstock-outputs/)
- [The conda-forge docs](https://conda-forge.org/docs/maintainer/adding_pkgs.html#the-staging-process)
- [Grayskull](https://github.com/conda-incubator/grayskull) - to automatically generate a recipe for packages on PyPI
- [The conda r skeleton helpers](https://github.com/bgruening/conda_r_skeleton_helper) - to automatically generate a recipe for packages on CRAN

# The requested package

- _Package name_: pyclibrary
- _Version (If a specific one is required)_:
- _URL_: https://pypi.org/project/pyclibrary & https://github.com/MatthieuDartiailh/pyclibrary
- _Available via (GitHub Releases, PyPI, etc.)_: Both
- _Additional comments_:

# Before submitting
- [X] Verify that the package does not already exist (Also check with _hyphens_ vs. _underscores_ and added identifiers like `python-` and `r-`).
- [X] Check that no previous [issue](https://github.com/conda-forge/staged-recipes/issues) exists and that no [PR](https://github.com/conda-forge/staged-recipes/pulls) is currently open.

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Closes #15962

### CODE REVIEW

**Comment by BastianZim** [comment]
CI is at https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=366048&view=results

**Comment by BastianZim** [comment]
@conda-forge/help-python Any idea why it cannot find the URL? When I click on it, it downloads it.

**Comment by BastianZim** [comment]
@conda-forge/help-python Would be ready for review, thanks!

File: recipes/PyClibrary/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+about:
+  home: http://github.com/MatthieuDartiailh/pyclibrary
+  summary: C binding automation
+  license: MIT
+  license_file: LICENSE
+  dev_url: https://pyclibrary.readthedocs.io/en/latest/
+
+extra:
+  recipe-maintainers:
+    - BastianZim
```
Comment (jakirkham) [inline recipes/PyClibrary/meta.yaml:45]: Feel free to add me as well 🙂

**Review by jakirkham — APPROVED** [review]
Thanks @BastianZim! LGTM 😄

File: recipes/PyClibrary/meta.yaml (hunk 1 above)
Comment (BastianZim) [reply recipes/PyClibrary/meta.yaml:46]: Wasn't sure how busy you are... 😄

**Comment by BastianZim** [comment]
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=366067&view=results

File: recipes/PyClibrary/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipes/PyClibrary/meta.yaml:46]: Always busy 😅

Thanks!

**Comment by jakirkham** [comment]
Thanks @BastianZim! 😄

File: recipes/PyClibrary/meta.yaml (hunk 1 above)
Comment (BastianZim) [reply recipes/PyClibrary/meta.yaml:46]: 😄 

My pleasure!

**Pull request merged**